luaui/Widgets/gui_gridmenu.lua

```diff
@@ -680,46 +660,53 @@ local function getActionHotkey(action)
 	return key
 end
 
+local function getGridKey(action)
+	local key = getActionHotkey(action)
+		or getActionHotkey(action .. ' builder')
+		or getActionHotkey(action .. ' factory')
+	return key
+end
+
 local function reloadBindings()
 	currentLayout = Spring.GetConfigString("KeyboardLayout", 'qwerty')
 
 	Cfgs.keyLayout = {{}, {}, {}}
 
 	for c=1,4 do
-		local cKey = getActionHotkey('gridmenu_category ' .. c)
+		local action = 'gridmenu_category ' .. c
+		local cKey = getGridKey(action)
 
 		if not cKey then
-			cKey = Cfgs.qwerty[1][c]
-			Spring.SendCommands("bind " .. cKey .. " gridmenu_category " .. c)
-			Spring.SendCommands("bind Shift+" .. cKey .. " gridmenu_category " .. c)
+			cKey = ''
+			Spring.Echo("Error, missing grid category keybind for action " .. action .. ", things may not function as expected")
 		end
 
 		Cfgs.categoryKeys[c] = cKey
 
 		for r=1,3 do
-			local key = getActionHotkey('gridmenu_key ' .. r .. ' ' .. c)
+			local action = 'gridmenu_key ' .. r .. ' ' .. c
+			local key = getGridKey(action)
 
 			if not key then
-				key = Cfgs.qwerty[r][c]
-				Spring.SendCommands("bind Any+" .. key .. ' gridmenu_key ' .. r .. ' ' .. c)
+				key = ''
+				Spring.Echo("Error, missing grid key bind for action " .. action .. ", things may not function as expected")
 			end
 
 			Cfgs.keyLayout[r][c] = key
 		end
 	end
 
-	local key = getActionHotkey('gridmenu_next_page')
+	local key = getGridKey('gridmenu_next_page')
 	if not key then
-		key = Cfgs.NEXT_PAGE_KEY
-		Spring.SendCommands('bind ' .. key .. ' gridmenu_next_page')
+		Spring.Echo("Error, missing grid key bind for next page, things may not function as expected")
 	end
 
 	Cfgs.NEXT_PAGE_KEY = key
 
-	key = getActionHotkey('gridmenu_prev_page')
+	key = getGridKey('gridmenu_prev_page')
```

Comment:
For `gridmenu_prev_page` bindings the `builder` or `factory` argument is invalid, just use `getActionHotkey`.